sigstore/_internal/trust.py

```diff
@@ -217,6 +220,102 @@ def __str__(self) -> str:
         return self.value
 
 
+class CertificateAuthority:
+    """
+    Certificate Authority used in a Trusted Root configuration.
+    """
+
+    def __init__(self, inner: _CertificateAuthority):
+        """
+        Construct a new `CertificateAuthority`.
+
+        @api private
+        """
+        self._inner = inner
+        self._leaf: Certificate | None = None
```

Comment:
Hmm, where is the concept of a "leaf" in a `CertificateAuthority` coming from? The message itself only defines `cert_chain` as being a list of root and intermediate CAs, so there's no "leaf" certificate per se (except in the sense that one of the intermediates is the last one).

I think we should drop this and its associated property:

```suggestion
```